FrontEnd/src/App.js

```diff
@@ -0,0 +1,155 @@
+
+const validateForm = (errors) => {
+    let valid = true;
+    Object.values(errors).forEach(
+      (val) => val.length > 0 && (valid = false)
+    );    
+    return valid;
+}
+
+class Register extends Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.recaptchaLoaded = this.recaptchaLoaded.bind(this);
+    this.verifyCallback = this.verifyCallback.bind(this);
+    
+    this.state = {
+      firstName: null,
+      email: null,
+      message: null,
+      isVerified: false,
+      errors: {
+        firstName: '',
+        lastName: '',
+        email: '',
+        message: ''
+      }
+    };
+  }
+
+  handleChange = (event) => {
+    event.preventDefault();
+    const { name, value } = event.target;
+    let errors = this.state.errors;
+
+    switch (name) {
+      case 'firstName': 
+        errors.firstName = value.length == 0 ? 'Please enter your first name' : '';
+        break;
+
+        case 'lastName': 
+        errors.lastName = value.length == 0  ? 'Please enter your last name'  : '';
+        break;
+      case 'email': 
+        errors.email = validEmailRegex.test(value) ? '' : 'Please enter a valid email address';
+        break;
+      case 'message': 
+        errors.message = value.length == 0 ? 'Please enter your message or question' : '';
+        break;
+      default:
+        break;
+    }
+
+    this.setState({errors, [name]: value});
+  }
+ 
+  recaptchaLoaded(){
+    console.log('recaptcha loaded');
+  }
+
+  verifyCallback = (response) => {
+    console.log("verifyCallback called and set to true");
+
+    if(response){      
+      this.setState({
+        isVerified: true
+      })
+      
+    }
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();    
+
+    if(!this.state.isVerified){
+      alert('Pleave verify that you are a human using the reCaptcha'); 
+      return;
+    }
+
+    if(validateForm(this.state.errors)  && 
+            this.state.firstName != null && 
+            this.state.lastName != null && 
+            this.state.email != null && 
+            this.state.message != null){    
+      console.info(this.state)
+      alert('Thank you! \nYour message has been successfully sent. We will contact you shortly!')   
+    }else{
+      alert('Oops! The form is missing information — please review.')
+    }
+  }
+
+  render() {
+    const {errors} = this.state;
+    return (
+      <div className='wrapper'>
+        <div className='form-wrapper'>
+          <h2>Contact Us</h2>
+          <form onSubmit={this.handleSubmit} noValidate>
+            <div className='firstName'>
+              <label htmlFor="firstName">First Name</label>
+              <input type='text' name='firstName' onChange={this.handleChange} Validate />
+              {errors.firstName.length > 0 && 
+                <span className='error'>{errors.firstName}</span>}
+            </div>
```

Comment:
Each of these form elements do a lot of the same things, like setting up a label, an input of a certain type, with a section for errors on the input.

Seems these elements could be refactored into one formElement component that you switch up with different props. What would that look like?

Reply:
Creating a base/parent object for these would have been significantly better than repeating the same bits of code over and over.  After all, this is where OOP excels.